raft/raft.go

```diff
@@ -1298,6 +1318,11 @@ func stepCandidate(r *raft, m pb.Message) error {
 	case myVoteRespType:
 		gr, rj, res := r.poll(m.From, m.Type, !m.Reject)
 		r.logger.Infof("%x has received %d %s votes and %d vote rejections", r.id, gr, m.Type, rj)
+		if m.Reject && m.Commit > r.raftLog.committed && r.raftLog.matchTerm(m.Commit, m.CommitTerm) {
```

Comment:
The combined conditions here feel subtle: If someone transmits a commit index/term pair with `Reject=false` it will be ignored. Why? I know that right now we don't send such an update, but if we started to in the future the versioning gets more complicated. It might be simpler (and remove some duplication) to hoist this check up to a higher level in `Step`: Whenever we receive a message with a commit and commit term, we advance to it if we can.